utils/common/src/main/java/org/apache/brooklyn/config/ConfigInheritance.java

```diff
@@ -20,31 +20,60 @@
 
 import java.io.Serializable;
 
+import org.apache.brooklyn.util.text.Strings;
+
 import com.google.common.annotations.Beta;
 
 @SuppressWarnings("serial")
 public abstract class ConfigInheritance implements Serializable {
 
-    public static final ConfigInheritance ALWAYS = new Always();
+    public enum InheritanceMode {
```

Comment:
Agreed. If we went back in time to review the very first pull request, then I'd change it to be simpler (rather than context-dependent behaviour). But now that we have the code, let's keep it "just in case".